src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizer.cs

```diff
@@ -249,15 +251,54 @@ private string GetResourceStreamName(CultureInfo culture)
             return resourceStreamName;
         }
 
+        private const string AssemblyElementDelimiter = ", ";
+
+        private string ApplyCultureToAssembly(string assemblyFullName, CultureInfo culture)
```

Comment:
This is a workaround for dotnet/coreclr#6123 so I'm hesitant to go too crazy with optimizing here but I suppose I could do `string.IndexOf`'s and `StringBuilder.Insert`'s instead unless you had something else in mind.